cpu/pic32/lib/pic32_pwm.c

```diff
@@ -0,0 +1,187 @@
+        int8_t minimum_prescaler_index = 7;                                                                         \
+        bool no_acceptable_prescaler = true;                                                                        \
+        while (minimum_prescaler_index != 0) {                                                                      \
+            uint32_t tmp = period >> 16;                                                                            \
+            uint32_t max_prescaler_period = pb_period * prescaler_values[minimum_prescaler_index];                  \
+            if (tmp >= max_prescaler_period)                                                                        \
+                break;                                                                                              \
+             else                                                                                                   \
+                no_acceptable_prescaler = false;                                                                    \
+            --minimum_prescaler_index;                                                                              \
+        }                                                                                                           \
+        if (no_acceptable_prescaler)                                                                                \
+          return -1;                                                                                                \
+                                                                                                                    \
+        /* 2. Compute PR values */                                                                                  \
+        uint16_t prs[8];                                                                                            \
+        int8_t i;                                                                                                   \
+        for (i = minimum_prescaler_index; i < 8; ++i)                                                               \
+            prs[i] = (period / (pb_period * prescaler_values[i])) - 1;                                              \
+                                                                                                                    \
+        /* 3. Compute error for each (PR,prescaler) */                                                              \
+        uint32_t errors[8];                                                                                         \
+        bool no_acceptable_pr = true;                                                                               \
+        const uint32_t max_error = period >> 5;   /* max_error= +-1.5625% of desired_period */                      \
+        for (i = minimum_prescaler_index; i < 8; ++i) {                                                             \
```

Comment:
Either switch this to the define used by the arrays, or get the array's size in the condition